src/Symfony/Component/Validator/ValidatorBuilder.php

```diff
@@ -295,7 +300,7 @@ public function getLoaders()
     /**
      * {@inheritdoc}
      */
-    public function getValidator()
+    public function getValidator(/*TranslatorInterface $translator = null*/)
```

Comment:
that looks wrong to me. This class is a builder, so passing one of the (optional) dependencies to the final building method makes this one very special compared to the other settings.

Reply:
I get that. That's the best I could think of for now. Any better suggestion?

Reply:
IMO, it should stay on the setter, like today

Reply:
I'd like to. How could be deal with BC then?

Reply:
Well, too bad that we haven't made this builder a `@final` class but an interface with an extendable implementation instead. That does not make sense. ValidatorBuilder is not an extension point (the interface is useless as it is not injected anywhere, and would forbid adding any method).
So for now, it would require using a new method as we cannot remove the typehint. to accept both.

Reply:
We can deprecate the interface and make the class final. That should do it.
In order to pass the type hint, we would need to make the new TranslatorInterface extend the old one when the component is found. Would that work for you?

Reply:
> new TranslatorInterface extend the old one

done, and make ValidatorBuilder final + deprecated its interface